paparazzi-gradle-plugin/src/test/projects/build-class-next-sdk/src/test/java/app/cash/paparazzi/plugin/test/BuildClassTest.kt

```diff
@@ -27,8 +27,8 @@ class BuildClassTest {
 
   @Test
   fun verifyFields() {
-    assertThat(Build.ID).isEqualTo("URD9.230712.002.A1")
-    assertThat(Build.DISPLAY).isEqualTo("sdk_phone_armv7-userdebug UpsideDownCakePrivacySandbox URD9.230712.002.A1 10628788 test-keys")
+    assertThat(Build.DISPLAY).isEqualTo("sdk_phone_armv7-userdebug UpsideDownCakePrivacySandbox URD9.231106.004.A2 11164158 test-keys")
+    assertThat(Build.ID).isEqualTo("URD9.231106.004.A2")
```

Comment:
How did truth/guava affect these?

Reply:
good point, i think just non-hermetic update from Google's side.  I can move into a separate PR.

Reply:
I'm happy either way, was just surprised.

Do I understand it correctly that this change depends on the Android SDK via compileSdk?

Reply:
Yes, but now this change is leading me to pull on a long thread haha.  When I push an empty commit, CI passes, but if i bump Truth, it fails....why?  Is our Gradle caching setup on CI busted in some way?

Reply:
Very likely
 * mandatory mention https://github.com/cashapp/paparazzi/pull/997
 * but also if something randomly reads a file from the SDK during `test` that must be also declared as an input, otherwise Gradle doesn't know.

Based on "empty commit" mention, it sounds like the second, assuming that when you push empty commit no test gets re-executed (which should be the case if every task is cached by gradle-build-action).

How this relates to truth upgrade is beyond me still though. You can try modifying this file in a non-meaningful way (e.g. add a comment) and see if it fails or passes.

Reply:
Did a deep search of the SDK and found that this is located in: `platforms\android-34\build.prop`, then noticed that in the Paparazzi codebase this is mentioned here:

https://github.com/cashapp/paparazzi/blob/aec3fa37941b5ac4b580820f9fee9efc12107a28/paparazzi/src/main/java/app/cash/paparazzi/internal/Renderer.kt#L112-L124

So the fix for this is to add all of those files as inputs to the test task, otherwise Gradle thinks that nothing has changed and load from cache.

Reply:
nice find!  

Ok, so PrepareResourcesTask will re-run when compileSdkVersion changes:
```
  abstract val compileSdkVersion: Property<String>
```

which will cause the test task to re-run again
```
        test.systemProperties["paparazzi.test.resources"] =
          writeResourcesTask.flatMap { it.paparazziResources.asFile }.get().path
```

the issue here is 99% of the time `$ANDROID_HOME/platforms/android-${compileSdkVersion}` will have static contents, but 1% (when using sandbox-level sdks), this will not be the case, as we're seeing here because build.prop is changed, but not being detected.

I'd hate to change the plugin to crawl through the entire platformDir checking for file content changes for the 1% test case, especially given initial findings that we might be caching huge payloads, so I think the "right fix" here is to disable caching for that specific plugin test.

Reply:
https://github.com/cashapp/paparazzi/pull/1211

Reply:
Another option -- if this comes up again -- would be to force the PrepareResourcesTask to rerun when compileSdkVersion doesn't parse to an integer value.